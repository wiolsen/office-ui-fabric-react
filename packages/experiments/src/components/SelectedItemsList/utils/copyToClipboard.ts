export function copyToClipboard(copyText: string): void {
  const copyInput = document.createElement('input') as HTMLInputElement;
  document.body.appendChild(copyInput);

  try {
    // Try to copy the text directly to the clipboard
    copyInput.value = copyText;
    copyInput.select();
    if (!document.execCommand('copy')) {
      // The command failed. Fallback to the method below.
      throw new Error();
    }
  } catch (err) {
    // no op
  } finally {
    document.body.removeChild(copyInput);
  }
}
