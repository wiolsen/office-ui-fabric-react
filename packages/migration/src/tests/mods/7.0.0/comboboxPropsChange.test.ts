import valueMigration from '../../../mods/7.0.0/comboboxValuePropChange';
import onChangeWarning from '../../../mods/7.0.0/comboboxOnChangedPropsChange';
import path from 'path';
import { runMigration } from '../utils';

describe('comboboxPropsChange', () => {
  it('transforms value to text', async () => {
    const result = await runMigration(valueMigration, path.join('7.0.0', 'combobox.tsx'));
    expect(result.warnings).toEqual([]);
    expect(result.contents).toMatchSnapshot();
  });

  it('warns for onChanged prop removal', async () => {
    const result = await runMigration(onChangeWarning, path.join('7.0.0', 'combobox.tsx'));
    expect(result.warnings.length).toEqual(1);
  });
});
