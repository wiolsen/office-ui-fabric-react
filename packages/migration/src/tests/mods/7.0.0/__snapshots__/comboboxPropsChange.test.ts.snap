// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comboboxPropsChange transforms value to text 1`] = `
"import { ComboBox, Fabric } from 'office-ui-fabric-react';

class ComboBoxControlledExample extends React.Component<{}, {}> {
  public render(): JSX.Element {
    return (
      <Fabric className={wrapperClassName}>
        <ComboBox
          selectedKey={state.selectedOptionKey}
          label=\\"Controlled single-select ComboBox (allowFreeform: T)\\"
          allowFreeform={true}
          autoComplete=\\"on\\"
          options={state.options}
          onChanged={this._onChange}
          onResolveOptions={this._getOptions}
          text={state.initialDisplayValue}
        />
      </Fabric>
    );
  }
}
"
`;
